### PR TITLE
Implemented arrow data getter / extraction

### PR DESCRIPTION
**Pull request by JohanMabille**

### CODE REVIEW

File: test/test_array.cpp [hunk 1]
```diff
@@ -72,6 +72,121 @@ namespace sparrow
...
+                ArrowArray ar{};
+                test::fill_schema_and_array<scalar_value_type>(sc, ar, size, offset, {});
+                array a(std::move(ar), std::move(sc));
+
+                ArrowSchema* sc_ptr = nullptr;
+                ArrowArray* ar_ptr = nullptr;
+                a.get_arrow_array(ar_ptr).get_arrow_schema(sc_ptr);
+
+                auto pa = primitive_array<scalar_value_type>(arrow_proxy(ar_ptr, sc_ptr));
+
```
Comment (Alex-PLACET) [inline test/test_array.cpp:143]: You have to call the release callback of both array and schema and delete both pointers.

File: test/test_array.cpp [hunk 2]
```diff
@@ -72,6 +72,121 @@ namespace sparrow
...
+                test::fill_schema_and_array<scalar_value_type>(sc, ar, size, offset, {});
+                array a(&ar, &sc);
+
+                ArrowSchema* sc_ptr = nullptr;
+                ArrowArray* ar_ptr = nullptr;
+                a.get_arrow_array(ar_ptr).get_arrow_schema(sc_ptr);
+
+                auto pa = primitive_array<scalar_value_type>(arrow_proxy(ar_ptr, sc_ptr));
+
+                CHECK_EQ(pa, pa_ctrl);
```
Comment (Alex-PLACET) [inline test/test_array.cpp:128]: You have to call the release callback of both array and schema and delete both pointers.

**Review by Alex-PLACET — CHANGES_REQUESTED** [review]

File: test/test_array.cpp (hunk 1 above)
Comment (JohanMabille) [reply test/test_array.cpp:143]: `ar_ptr` points to `&ar` and `sc_ptr` points to `&sc` so they should not be deleted (`ar` and `sc` are allocated on the stack and will be destroyed at the end of the scope). The release methods of these structures must still be called though, I find it more intuitive to call them on the original structure (i.e. `ar` and `sc`).

File: test/test_array.cpp (hunk 2 above)
Comment (JohanMabille) [reply test/test_array.cpp:128]: ar_ptr points to &ar and sc_ptr points to &sc so they should not be deleted (ar and sc are allocated on the stack and will be destroyed at the end of the scope). The release methods of these structures must still be called though, I find it more intuitive to call them on the original structure (i.e. ar and sc).

Comment (Alex-PLACET) [reply test/test_array.cpp:129]: You are right for this one the ar_ptr and sc_ptr are pointers over ar and sc.

File: test/test_array.cpp (hunk 1 above)
Comment (Alex-PLACET) [reply test/test_array.cpp:150]: I mixed with extrat. You are right, a will release the array and schema.

**Review by Alex-PLACET — APPROVED** [review]

**Pull request merged**